src/sql/workbench/services/connection/browser/connectionManagementService.ts

```diff
@@ -1045,6 +1045,9 @@ export class ConnectionManagementService extends Disposable implements IConnecti
 			if (this._connectionStatusManager.isDefaultTypeUri(info.ownerUri)) {
 				this._connectionGlobalStatus.setStatusToConnected(info.connectionSummary);
 			}
+			if (info.isSupportedVersion === false) {
+				this._notificationService.warn(nls.localize('connection.unsupportedServerVersionWarning', "The server version is no longer supported. You may still connect to it but some features in Azure Data Studio might not work as expected."));
```

Comment:
Would be good to provide a link to the versions that are supported.

Also for stuff like this I think we should have a "Don't show again for this connection" or something similar

Reply:
The text `The server version is no longer supported` should probably be something like `Azure Data Studio does not support this server version` to make clear that it's ADS that doesn't support the server version, the server version may still be under long-term support in general (through SSMS, sqlcmd, etc.).

Agree we need an option to "do not show again".  I think it's fine for the don't show again flag to be global vs. per connection, though per connection may be better if it's not too complicated to track.

Reply:
added a new property to allow connection provider to provide additional messages, for example the link to the web page that describes the supported versions and the alternate tools for unsupported versions.

added an option for users to completely disable the warning message. otherwise I am still making sure to only show it once for a connection within the lifespan of ADS, OE, Dashboard, Query Editor and other places will establish new connections that will trigger the warning. I am not using the connection's ownerUri because it might have the prefix added like: dashboard, connection, notebook, and query editor is worse, it is 'untitled: sqlquery_1'. I had to use the actual unique key for it.